core/modules/qdisp/JobQuery.cc

```diff
@@ -136,12 +138,19 @@ bool JobQuery::cancel() {
         std::lock_guard<std::recursive_mutex> lock(_rmutex);
         // If _queryRequestPtr is not nullptr, then this job has been passed to xrootd and
         // cancellation is complicated.
+        bool cancelled = false;
         if (_queryRequestPtr != nullptr) {
             LOGS_DEBUG(getIdStr() << " cancel QueryRequest in progress");
-            _queryRequestPtr->cancel();
-        } else {
+            if (_queryRequestPtr->cancel()) {
+                LOGS_DEBUG(getIdStr() << " cancelled by QueryRequest");
```

Comment:
Re LOGS vs LOGS_DEBUG - I think our preferred way is to use specific logger instead of dumping it all to a root logger, so `LOGS(_log, ...)` wins here. No opinion on `_idStr` vs `getIdStr()`.